packages/sitecore-jss-react/src/components/ComponentLibraryLayout.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import { Placeholder } from './Placeholder';
+import {
+  ComponentRendering,
+  EDITING_COMPONENT_ID,
+  EDITING_COMPONENT_PLACEHOLDER,
+  LayoutServiceData,
+} from '@sitecore-jss/sitecore-jss/layout';
+import {
+  addComponentUpdateHandler,
+  COMPONENT_LIBRARY_READY_MESSAGE,
+} from '@sitecore-jss/sitecore-jss/component-library';
+
+export const ComponentLibraryLayout = (layoutData: LayoutServiceData): JSX.Element => {
+  const { route } = layoutData.sitecore;
+  const [renderSwitch, setRenderSwitch] = useState(false);
+  const rootComponent = route?.placeholders[EDITING_COMPONENT_PLACEHOLDER][0] as ComponentRendering;
+  // useEffect may execute multiple times - but we only want to subscribe and fire the events once
+  let componentReady = false;
+  let [updateListenerReady] = useState(false);
+
+  useEffect(() => {
+    // useEffect will fire when components are ready - and we inform the whole wide world of it too
+    if (!componentReady) {
+      componentReady = true;
+      window.postMessage(COMPONENT_LIBRARY_READY_MESSAGE, '*');
+    }
+    if (!updateListenerReady) {
+      addComponentUpdateHandler(rootComponent, () => setRenderSwitch(!renderSwitch));
```

Comment:
Customer solutions might do weird things. There is no need to find what is exact scenario they might do to send an updated layout. The problem here that this component as a separate unit works unexpectedly for a common operation, which is just updating a prop.

Reply:
@stasmaxymov  I resolved and closed other comments, but left this one open.
Overall the approach I went with is what you see now :) I thought about best ways to pass custom markup/layouts into ComponentLibraryLayout - but it seems more trouble than it's worth. There's no good way to enforce a contract for layouts passed as props, and if we do it would be too easy to break the whole thing.
An acceptable middlepoint I found is to allow devs modify layout in /component-library/render